tests/load/pipeline/test_filesystem_pipeline.py

```diff
@@ -172,3 +185,64 @@ def some_source():
     with open(other_data_files[0], "rb") as f:
         table = pq.read_table(f)
         assert table.column("value").to_pylist() == [1, 2, 3, 4, 5]
+
+
+TEST_LAYOUTS = (
+    "{schema_name}/{table_name}/{load_id}.{file_id}.{ext}",
+    "{schema_name}.{table_name}.{load_id}.{file_id}.{ext}",
+    "{table_name}88{load_id}-u-{file_id}.{ext}",
+    "{table_name}/{curr_date}/{load_id}.{file_id}.{ext}{timestamp}",
+    "{table_name}/{YYYY}-{MM}-{DD}/{load_id}.{file_id}.{ext}",
+    "{table_name}/{YYYY}-{MMM}-{D}/{load_id}.{file_id}.{ext}",
+    "{table_name}/{DD}/{HH}/{m}/{load_id}.{file_id}.{ext}",
+    "{table_name}/{D}/{HH}/{mm}/{load_id}.{file_id}.{ext}",
+    "{table_name}/{timestamp}/{load_id}.{file_id}.{ext}",
+    "{table_name}/{load_package_timestamp}/{d}/{load_id}.{file_id}.{ext}",
+    (
+        "{table_name}/{YYYY}/{YY}/{Y}/{MMMM}/{MMM}/{MM}/{M}/{DD}/{D}/"
+        "{HH}/{H}/{ddd}/{dd}/{d}/{Q}/{timestamp}/{curr_date}/{load_id}.{file_id}.{ext}"
+    ),
+)
+
+
+@pytest.mark.parametrize("layout", TEST_LAYOUTS)
+def test_filesystem_destination_extended_layout_placeholders(layout: str) -> None:
+    data = load_json_case("simple_row")
+    call_count = 0
+
+    def counter(value: Any) -> Callable[..., Any]:
+        def count(*args, **kwargs) -> Any:
+            nonlocal call_count
+            call_count += 1
+            print("call\n", value)
+            return value
+
+        return count
+
+    extra_placeholders = {
+        "who": "marcin",
+        "action": "says",
+        "what": "no potato",
+        "func": counter("lifting"),
+        "woot": "woot-woot",
+        "hiphip": counter("Hurraaaa"),
+    }
+    layout_normalized = snake_case_naming_convention.normalize_path(layout)
+    os.environ["DESTINATION__FILESYSTEM__BUCKET_URL"] = "file://_storage"
+    pipeline = dlt.pipeline(
+        pipeline_name=f"test_{layout_normalized[2:8]}_pipeline",
+        destination=filesystem(
+            layout=layout,
+            extra_placeholders=extra_placeholders,
+            kwargs={"auto_mkdir": True},
+            current_datetime=counter(pendulum.now()),
+        ),
+    )
+    pipeline.run(
+        dlt.resource(data, name="simple_rows"),
+        write_disposition="append",
+    )
+
+    # 6 is because simple_row contains two rows
+    # and in this test scenario we have 3 callbacks
+    assert call_count >= 6
```

Comment:
this is actually pretty cool. but again: are you able to compare paths from fsspec.ls to a reference? you know the layout, you know the extra and standard placeholders and from unit tests you know that create_path works well. so do it

Reply:
That would be cool to list all files and compare paths

Reply:
Done.